packages/core/src/lib/markdown-it/highlight/HighlightRule.ts

```diff
@@ -61,56 +66,34 @@ export class HighlightRule {
     return atLineNumber;
   }
 
-  applyHighlight(line: string, lineNumber: number) {
-    // Applied rule is the first component until deduced otherwise
+  getHighlightType(lineNumber: number): {
+    highlightType: HIGHLIGHT_TYPES,
+    bounds: Array<[number, number]> | null
+  } {
     let [appliedRule] = this.ruleComponents;
-
     if (this.isLineRange()) {
-      // For cases like 2[:]-3 (or 2-3[:]), the highlight would be line highlight
-      // across all the ranges
```

Comment:
isn't this still applicable?

Reply:
I will add them back, for the applied rule being first component I felt the comment was redundant but yap anyways